test/test_dht.cpp

```diff
@@ -672,7 +677,7 @@ TORRENT_TEST(dht)
 	// 50 downloaders and 50 seeds
 	for (int i = 0; i < 100; ++i)
 	{
-		source = udp::endpoint(rand_v4(), 6000);
+		source = udp::endpoint((*rand_addr)(), 6000);
```

Comment:
you don't actually have to dereference the function pointer to call it. If you don't like that special case, you could also just make it a function reference (which probably is better anyway, since we don't need to be able to re-bind this variable to any other function anyway)